File: android/src/main/java/com/infopro/sslchecker/SSLCheckerPlugin.java
package com.infopro.sslchecker;

import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import org.json.JSONException;

@CapacitorPlugin(name = "SSLChecker")
public class SSLCheckerPlugin extends Plugin {

    private SSLChecker implementation = new SSLChecker();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void verify(PluginCall call) {
        String url = call.getString("url");
        JSArray fingerprints = call.getArray("fingerprints");

        JSObject ret = new JSObject();
        ret.put("value", implementation.verify(url, fingerprints));
        call.resolve(ret);
    }
}

File: ios/Sources/SSLCheckerPlugin/SSLCheckerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SSLCheckerPlugin)
public class SSLCheckerPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "SSLCheckerPlugin"
    public let jsName = "SSLChecker"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "verify", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = SSLChecker()

    @objc func verify(_ call: CAPPluginCall) {
        let serverUrl = call.getString("url") ?? ""
        let fingerprints = (call.getArray("fingerprints", String.self) ?? []) as NSArray

        
        call.resolve([
            "value": implementation.verify(serverUrl: serverUrl, allowedFingerprints: fingerprints)
        ])
    }
}
